src/app/components/HighlightedText/text-matching.ts

```diff
@@ -0,0 +1,106 @@
+import { NormalizerOptions, normalizeTextForSearch } from './text-normalization'
+export type { NormalizerOptions } from './text-normalization'
+
+/**
+ * Store info about where did we found the pattern inside the corpus
+ */
+export interface MatchInfo {
+  searchText: string
+  startPos: number
+}
+
+/**
+ * Identify pattern matches within a corpus, also considering normalization
+ */
+export const findTextMatch = (
+  rawCorpus: string | null | undefined,
+  search: (string | undefined)[],
+  options: NormalizerOptions = {},
+): MatchInfo => {
+  let matchStart: number
+  let normalizedPattern: string
+  const normalizedCorpus = normalizeTextForSearch(rawCorpus || '', options)
+  const matches: MatchInfo[] = search
+    .filter(s => !!s && s.toLowerCase)
+    .map(rawPattern => {
+      normalizedPattern = normalizeTextForSearch(rawPattern!, options)
+      // console.log(`Will search for "${normalizedPattern}" in "${normalizedCorpus}"...`);
+      matchStart = normalizedCorpus.indexOf(normalizedPattern)
+      return matchStart !== -1
+        ? {
+            searchText: rawPattern,
+            startPos: matchStart,
+          }
+        : undefined
+    })
+    .filter(m => !!m)
+    .map(m => m as MatchInfo) // This last line is only here to make TS happy
+  return matches[0]
+}
+
+export interface CutAroundOptions extends NormalizerOptions {
+  /**
+   * What should be the length of the fragment delivered, which
+   * has the pattern inside it?
+   *
+   * The default value is 80.
+   */
+  fragmentLength?: number
+
+  debug?: boolean
+}
+
+/**
+ * Return a part of the corpus that contains the match to the pattern, if any
+ *
+ * If either the corpus or the pattern is undefined or empty, undefined is returned.
+ * If there is no match, undefined is returned.
+ *
+ * If there is a match, but the corpus is at most as long as the desired fragment length,
+ * the whole corpus is returned.
+ *
+ * If there is a match, and the corpus is longer than the desired fragment length,
+ * then a part of a corpus is returned, so that the match is within the returned part,
+ * around the middle.
+ */
+export function cutAroundMatch(
```

Comment:
Currently, yes. It's useful for use cases when (for example) you are searching in longer text samples, and showing search results in a table, and you want to showcase the exact location and immediate vicinity of each match. 

Since we don't do that right now, removed.